packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/repository.ts

```diff
@@ -762,6 +772,247 @@ export class SavedObjectsRepository implements ISavedObjectsRepository {
     );
   }
 
+  /**
+   *  {@inheritDoc ISvedObjectsRepository.bulkDelete}
+   */
+  async bulkDelete(
+    objects: SavedObjectsBulkDeleteObject[],
+    options: SavedObjectsBulkDeleteOptions
+  ): Promise<SavedObjectsBulkDeleteResponse> {
+    const { refresh = DEFAULT_REFRESH_SETTING, force } = options;
+    const namespace = normalizeNamespace(options.namespace);
+    let bulkGetRequestIndexCounter = 0;
+
+    const expectedBulkGetAllDocsWithNamespaceChecksFlag =
```

Comment:
Please ignore these overly long, descriptive names. They will be renamed in future commits.